cep-20.2.md

```diff
@@ -0,0 +1,711 @@
+  include_recipe: bool (defaults to true)
+
+  # environment variables to either pass through to the script environment or set
+  script_env: [env_vars]
+
+  # A run export adds a dependency to the run requirements of a package if listed in build or host dependencies
+  run_exports: [MatchSpec] OR {strong: [MatchSpec], weak: [MatchSpec], strong_constrains: [MatchSpec], weak_constrains: [MatchSpec], noarch: [MatchSpec]}
+
+  # Allow linking against libraries that are not in the run requirements
+  missing_dso_whitelist: [glob]
+
+  # Allow runpath / rpath to point to these locations outside of the environment
+  runpath_whitelist: [glob]
+
+  # This is only used in the pip feedstock.
+  disable_pip: bool (defaults to false)
+
+  # merge the build and host environments (used in many R packages on Windows)
+  merge_build_host: bool
+
+  # ignore run exports by name
+  ignore_run_exports: list
+
+  # ignore run exports coming from the specified packages
+  ignore_run_exports_from: list
+
+  # copies fn[.bat/.sh] to the appropriate location, adds `.bat` or `.sh` to the filename
+  pre-link: string
+  post-link: string
+  pre-unlink: string
+
+  # error out when overdepending
+  error_overdepending: bool (defaults to ?)
+  # error out when overlinking
+  error_overlinking: bool (defaults to ?)
+
+  # REMOVED:
+  # noarch_python: bool
+  # features: list
+  # msvc_compiler: str
+  # requires_features: dict
+  # provides_features: dict
+  # preferred_env: str
+  # preferred_env_executable_paths: list
+
+  # marked as "still experimental"
+  # pin_depends: Enum<"record" | "strict">
+  # overlinking_ignore_patterns: [glob]
+
+  # defaults to patchelf (only cudatoolkit is using `lief` for some reason)
+  # rpaths_patcher: None
+```
+
+## Source section
+
+```yaml
+source: [SourceElement]
+```
+
+where the different source elements are defined as follows.
+
+### URL source
+
+```yaml
+# url pointing to the source tar.gz|zip|tar.bz2|...
+url: url
+# destination folder in work directory
+folder: path
+# hash of the file
+sha256: hex string
+# absolute or relative path from recipe file
+patches: [path]
+# removed keys
+# fn: string
+# md5: hex string
+# sha1: hex string
+```
+
+### Local source
+
+```yaml
+# file, absolute or relative to recipe file
+path: path
+# destination folder
+folder: path
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Git source
+
+```yaml
+# URL to the git repository or path to local git repository
+git_url: url | path
+# revision to checkout to (commit or tag)
+git_rev: string
+git_depth: signed integer (defaults to -1 -> not shallow)
+```
+
+### Removed source definitions
+
+SVN and HG (mercury) source definitions are removed as they are not relevant anymore.
+
+## Requirements section
+
+```yaml
+requirements:
+  # build time dependencies, in the build_platform architecture
+  build: [MatchSpec]
+  # dependencies to link against, in the target_platform architecture
+  host: [MatchSpec]
```

Comment:
Echoing the discussion in Element, there's an opportunity here to rename `host` to `target`, and align it with `target_platform`, which IMO better describes the intent.

It's true that it conflates a bit with the [GCC terminology](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html), but in all honesty, the conda ecosystem is mostly non-compiler packages so I am not sure where the value in keeping this relic of autotools is.

Reply:
Without being familiar with GCC terminology, but being familiar with conda recipes, I find build/target/run to be more descriptive of what's going on, especially when those terms are appended with "environment".

Reply:
Please don't change it to `target`. It needlessly confuses things. `target_platform` variable is wrong IMO. It should have been `host_platform.
We need better docs on `build, host, run`.

Reply:
I think `target` is way way way easier for people to catch on to and that accepting some ambiguity in the full Canadian cross case is preferable to having this large hurdle to understanding.

That said, we need good docs in either case. I've picked up a lot of bits and pieces over the years, but I've held off on writing docs because of the inevitable "you're wrong about X".

To illustrate what I mean, the most understandable explanation from my POV for the distinction between build and host is the basic cross-compilation case, where the distinction is easy to get (e.g. we have x64 agents to build, but the final package needs to run on <arch>; the build-time dependencies for that architecture are in `host`, etc etc). That explanation is not 100% accurate, but way more effective for transmitting the essentials. However I've never had the energy to face that discussion.

In short: +1 for target

Reply:
I'm -100 for target. There are two different nomenclatures.
1. autotools and conda - build and host
2. cmake etc. - host and target

Now, this suggestion is to introduce a third
3. build and target

Please don't add a third into the mix. Somewhat related meme: https://xkcd.com/927/